src/server.ts

```diff
@@ -1,27 +1,42 @@
-import express, { Request, Response, ErrorRequestHandler } from 'express'
-import { Person } from './types'
+import express, { Request, Response, ErrorRequestHandler } from 'express';
+import { Person } from './types';
+import fetch from 'node-fetch';
+import _, { omitBy, isEmpty } from 'lodash';
 const app = express();
 const port = 80;
-const request = require('request');
-const _ = require('lodash')
 
-app.get('/', (req:Request, res:Response) => {
-  request('https://randomuser.me/api/', function (error:ErrorRequestHandler, response:Response, body:Request["body"]) {
-    if(!error && response.statusCode == 200) {
-      body = JSON.parse(body);
-      const person:Person = body.results[0];
-      person["jobs"] = "fullstack dev at sssense";
-      if(!req.query.field) {
-        res.send(person);
-        return;
-      }
-      const fields = (req.query.field as string).split(",");
-      const filtered = _.pick(person, fields);
-      res.send(filtered);
-    }
-  })
+app.get('/', async (req:Request, res:Response) => {
+  const params = getParameters(req);
+  const url = getURL(params, req);
+  const response = await fetch(url);
+  const data = await response.json();
+  const person:Person = data.results[0];
+  if(req.query.field && (req.query.field as string).includes("jobs")) {
+    person["jobs"] = "fullstack dev at ssense";
+  }
```

Comment:
remember that we need `jobs` too even if we dont pass any fields, which may not be the case here